channels/audin/server/audin.c

```diff
@@ -142,16 +142,30 @@ static UINT audin_server_recv_formats(audin_server_context* context, wStream* s,
 		WLog_Print(audin->log, WLOG_WARN,
 		           "cbSizeFormatsPacket is invalid! Expected: %u Got: %zu. Fixing size",
 		           pdu.cbSizeFormatsPacket, Stream_GetPosition(s));
-		pdu.cbSizeFormatsPacket = Stream_GetPosition(s);
+		const size_t pos = Stream_GetPosition(s);
+		if (pos > UINT32_MAX)
+		{
+			error = ERROR_INVALID_PARAMETER;
```

Comment:
This looks weird, because the same is not applied to the previous condition here (`pdu.cbSizeFormatsPacket != Stream_GetPosition(s)`).

In any case, it is not a fatal error to hit the `pdu.cbSizeFormatsPacket != Stream_GetPosition(s)` condition. The `cbSizeFormatsPacket` value is more for convenience purpose, since we can derive this value here too.
So, this should probably incorporate the warning message too.

Reply:
added a log